### PR TITLE
Fix #904, Required regenerate and save API (req.session.regenerate is not a function since upgrade to 0.6.0)

### LINKED ISSUE #904: Sessions do not always have regenerate() and save() cauing a fault
I am the author of MeshCentral and use `cookie-session` along with Password. A recent [commit in Password](https://github.com/jaredhanson/passport/commit/7e9b9cf4d7be02428e963fc729496a45baeea608) is [causing crashes](https://github.com/Ylianst/MeshCentral/issues/4034) because `cookie-session` does not have a regenerate() or save() method.

See this issue: https://github.com/expressjs/cookie-session/issues/166

Could Passport revert to the old style of setting session values unless these methods exist? Thanks.

### PR DESCRIPTION
**Pull request by joeyguerra**
# Background RE: #904 

Using `passport@v0.6.0`, `passport-oauth2@v1.6.1` with the `cookie-session@v2.0.0` module to login with Github, the application throws the following error message:

```
TypeError: req.session.regenerate is not a function at SessionManager.logIn
```

# Feature

Utilize the Delegate Pattern and default to an empty implementation so that the code does not throw an error.

# Checklist

- [ x ] I have read the [CONTRIBUTING](https://github.com/jaredhanson/passport/blob/master/CONTRIBUTING.md) guidelines.
- [ x ] I have added test cases which verify the correct operation of this feature or patch.
- [ x ] I have added documentation pertaining to this feature or patch.
- [ x ] The automated test suite (`$ make test`) executes successfully.
- [ x ] The automated code linting (`$ make lint`) executes successfully. (NOTE: Lint errors are with code not touched by this PR)

### CODE REVIEW

File: lib/sessionmanager.js [hunk 1]
```diff
@@ -22,14 +33,13 @@ SessionManager.prototype.logIn = function(req, user, options, cb) {
   
   var self = this;
   var prevSession = req.session;
-  
+
   // regenerate the session, which is good practice to help
   // guard against forms of session fixation
-  req.session.regenerate(function(err) {
+  this._delegate.regenerate(req, function(err) {
```
Comment (imartinezmorales-loom) [inline lib/sessionmanager.js:39]: You should change this to `self` here - I still experiencing issues with the `regenerate` and `save` methods identifying as undefined when using this implementation.

Comment (raulrene) [reply lib/sessionmanager.js:39]: I couldn't get it working with either `self` or `this`

Comment (raulrene) [reply lib/sessionmanager.js:39]: @imartinezmorales-loom this code worked for me with `cookie-parser` `0.6.0`. It works both with `self` or `this`

**Review by raulrene — APPROVED** [review]
@jaredhanson Any chance you can take a look over this?

**Comment by maicongodinho** [comment]
Is there a possibility of this being merged in the near future?

File: lib/sessionmanager.js (hunk 1 above)
Comment (joeyguerra) [reply lib/sessionmanager.js]: I made the change. thanks for the review.

**Pull request closed without merging**